### PR TITLE
 sock_util: add interface descriptor parsing to str2ep 

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
When using `sock_udp_str2ep()` there was previously no way to parse the network interface when provided in the string, e.g. `[fe80::1%3]:5241`. This adds this feature.

### Testing procedure
I provided unittests for the new cases so just run
```sh
make -C tests/unittests/ tests-sock_util flash-only test
```
on a board of your liking.

### Issues/PRs references
See https://github.com/RIOT-OS/RIOT/issues/14035#issuecomment-627359833

### CODE REVIEW

**Comment by miri64** [comment]
Ping?

File: sys/net/sock/sock_util.c [hunk 1]
```diff
@@ -147,6 +150,45 @@ int sock_urlsplit(const char *url, char *hostport, char *urlpath)
...
+    for (netifend = netifstart; *netifend && *netifend != ']';
+         netifend++);
+    netiflen = netifend - netifstart;
+    if (!*netifend || (netiflen >= NETIF_STR_LEN) || (netiflen == 0)) {
+        /* no netif found, bail out */
+        return -EINVAL;
+    }
+    strncpy(netifbuf, netifstart, netiflen);
+    int netif = strtol(netifbuf, NULL, 10);
+    if ((netif > UINT16_MAX) || (netif < 0)) {
```
Comment (benpicco) [inline sys/net/sock/sock_util.c:185]:
```suggestion
    if ((netif > INT16_MAX) || (netif < 0)) {
```

Comment (miri64) [reply sys/net/sock/sock_util.c:185]: but the `netif` member of a `sock_udp_ep_t` is `uint16_t`...

Comment (benpicco) [reply sys/net/sock/sock_util.c:185]: > ```
> [01m[Ksock_util.c:185:16:[m[K [01;31m[Kerror: [m[Kcomparison between signed and unsigned integer expressions [-Werror=sign-compare]
>      if ((netif > UINT16_MAX) || (netif < 0)) {
> ```

Comment (miri64) [reply sys/net/sock/sock_util.c:185]: So I should cast to the smaller scope? I think it makes more sense to cast `netif` to `(unsigned)` ;-).

**Comment by benpicco** [comment]
Please squash!

**Pull request merged**